plugins/orbslam2_plugin/orbslam2_task_samplers.py

```diff
@@ -0,0 +1,18 @@
+from typing import Optional
+
+from core.base_abstractions.task import TaskSampler
+from plugins.robothor_plugin.robothor_tasks import PointNavTask
+from plugins.robothor_plugin import robothor_task_samplers
+
+
+class PointNavDatasetTaskSampler(robothor_task_samplers.PointNavDatasetTaskSampler):
+
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self.orbslam_sensor = next(s for s in self.sensors if s.uuid == kwargs['orbslam_uuid'])
+
+    def next_task(self, force_advance_scene: bool = False) -> Optional[PointNavTask]:
+        task = super().next_task(force_advance_scene)
+        if task is not None:
+            self.orbslam_sensor.reset(self.env.agent_state())
+        return task
```

Comment:
I understand the inclination to do things this way but the downside is that the sensor now only works with this task sampler. Can you instead reset the `orbslam_sensor` whenever `task.num_steps_taken() == 0` in the `get_observation` call?

Reply:
Great! This is what I wanted to do, but didn't know how.